snuba/datasets/dataset_schemas.py

```diff
@@ -24,7 +24,7 @@ def __init__(
     def get_read_schema(self) -> Schema:
         return self.__read_schema
 
-    def get_write_schema(self) -> Schema:
+    def get_write_schema(self) -> TableSchema:
```

Comment:
This change makes it possible for the type checker to enforce only writable schemas are provided to code that needs to write.